src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -0,0 +1,512 @@
+    private final LongUnaryOperator supplyInitialId;
+    private final LongUnaryOperator supplyReplyId;
+    private final LongSupplier supplyTraceId;
+
+    private final Long2ObjectHashMap<AmqpServerConnection> correlations;
+    private final Long2ObjectHashMap<AmqpServerSession> sessions;
+    private final MessageFunction<RouteFW> wrapRoute;
+
+    private final BufferPool bufferPool;
+
+    public AmqpServerFactory(
+        AmqpConfiguration config,
+        RouteManager router,
+        MutableDirectBuffer writeBuffer,
+        BufferPool bufferPool,
+        LongUnaryOperator supplyInitialId,
+        LongUnaryOperator supplyReplyId,
+        LongSupplier supplyTraceId)
+    {
+        this.router = requireNonNull(router);
+        this.writeBuffer = requireNonNull(writeBuffer);
+        this.bufferPool = bufferPool;
+        this.supplyInitialId = requireNonNull(supplyInitialId);
+        this.supplyReplyId = requireNonNull(supplyReplyId);
+        this.supplyTraceId = requireNonNull(supplyTraceId);
+        this.correlations = new Long2ObjectHashMap<>();
+        this.sessions = new Long2ObjectHashMap<>();
+        this.wrapRoute = this::wrapRoute;
+    }
+
+    @Override
+    public MessageConsumer newStream(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length,
+        MessageConsumer throttle)
+    {
+        final BeginFW begin = beginRO.wrap(buffer, index, index + length);
+        final long streamId = begin.streamId();
+
+        MessageConsumer newStream = null;
+
+        if ((streamId & 0x0000_0000_0000_0001L) != 0L)
+        {
+            newStream = newInitialStream(begin, throttle);
+        }
+        else
+        {
+            newStream = newReplyStream(begin, throttle);
+        }
+        return newStream;
+    }
+
+    private MessageConsumer newInitialStream(
+        final BeginFW begin,
+        final MessageConsumer sender)
+    {
+        final long routeId = begin.routeId();
+        final long initialId = begin.streamId();
+
+        final MessagePredicate filter = (t, b, o, l) -> true;
+        final RouteFW route = router.resolve(routeId, begin.authorization(), filter, this::wrapRoute);
+        MessageConsumer newStream = null;
+
+        if (route != null)
+        {
+            final AmqpServerConnection connection = new AmqpServerConnection(sender, routeId, initialId);
+            correlations.put(supplyReplyId.applyAsLong(initialId), connection);
+            newStream = connection::onNetwork;
+        }
+        return newStream;
+    }
+
+    private MessageConsumer newReplyStream(
+        final BeginFW begin,
+        final MessageConsumer sender)
+    {
+        final long replyId = begin.streamId();
+        final AmqpServerConnection connection = correlations.remove(replyId);
+        MessageConsumer newStream = null;
+        if (connection != null)
+        {
+            newStream = connection::onNetwork;
```

Comment:
AMQP reply streams for the server will be application streams, not network streams, so this should be left as a `TODO` with `newStream` remaining `null` for now.